sdk/openai/azure-ai-openai/src/samples/java/com/azure/ai/openai/usage/ChatCompletionsWithYourData.java

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.ai.openai.usage;
+
+import com.azure.ai.openai.OpenAIClient;
+import com.azure.ai.openai.OpenAIClientBuilder;
+import com.azure.ai.openai.models.AzureChatExtensionConfiguration;
+import com.azure.ai.openai.models.AzureChatExtensionType;
+import com.azure.ai.openai.models.AzureCognitiveSearchChatExtensionConfiguration;
+import com.azure.ai.openai.models.ChatChoice;
+import com.azure.ai.openai.models.ChatCompletions;
+import com.azure.ai.openai.models.ChatCompletionsOptions;
+import com.azure.ai.openai.models.ChatMessage;
+import com.azure.ai.openai.models.ChatRole;
+import com.azure.core.credential.AzureKeyCredential;
+import com.azure.core.util.BinaryData;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Sample demonstrates the "Azure OpenAI on your data" feature.
+ */
+public class ChatCompletionsWithYourData {
```

Comment:
should we add an async sample as well?

Reply:
Yeah, I think it is useful for us to show the different call paradigms and even usage scenarios (e.g. with search index versus without). When I was adding this one, I couldn't help but feel a growing overlap of our "usage" samples with our unit tests - I think it kind of "waters down" the utility of each, if that makes sense. I'll dedicate some time next week to sorting this out (and when I do, I'll make sure we have coverage for things like async vs sync like you mentioned)